docs/aws/audit/elbmonitoring/rules/elb_web_tier_health_check.mdx

```diff
@@ -24,6 +24,203 @@ HIPAA
+
+2. **List all the Load Balancers**: Use the following AWS CLI command to list all the load balancers in your account.
+
+    ```
+    aws elbv2 describe-load-balancers
+    ```
+
+    This command will return a JSON output with details of all the load balancers.
+
+3. **Describe Load Balancer Attributes**: For each load balancer, you need to check its attributes. You can do this using the following command:
+
+    ```
+    aws elbv2 describe-load-balancer-attributes --load-balancer-arn <value>
+    ```
+
+    Replace `<value>` with the ARN of your load balancer. This command will return a JSON output with the attributes of the specified load balancer.
+
+4. **Check the Health Check Configuration**: In the JSON output, look for the `HealthCheck` attribute. This attribute contains the health check configuration for the load balancer. Make sure that the `HealthCheckPath` is set to the path where your application's health check page is located, `HealthCheckIntervalSeconds` is set to the time interval for health checks, and `HealthCheckTimeoutSeconds` is set to the amount of time to wait when receiving a response from the health check.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install the necessary AWS SDK for Python (Boto3) in your environment. Boto3 allows you to directly create, update, and delete AWS services from your Python scripts.
+
+```python
+pip install boto3
+```
+
+2. Import the necessary libraries and establish a session with AWS using your access and secret keys.
+
+```python
+import boto3
+
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
+    region_name='us-west-2'
+)
+```
+
+3. Create an ELB client using the session.
+
+```python
+elb = session.client('elb')
+```
+
+4. Now, you can use the `describe_load_balancers` function to get a list of all your load balancers. Then, iterate over each load balancer and use the `describe_health_check` function to get the health check configuration. Check if the target is HTTP or HTTPS, if not, it's a misconfiguration.
+
+```python
+load_balancers = elb.describe_load_balancers()
+for lb in load_balancers['LoadBalancerDescriptions']:
+    health_check = elb.describe_health_check(LoadBalancerName=lb['LoadBalancerName'])
+    if not health_check['HealthCheck']['Target'].startswith('HTTP') and not health_check['HealthCheck']['Target'].startswith('HTTPS'):
+        print(f"Misconfiguration detected in ELB: {lb['LoadBalancerName']}")
+```
+
+This script will print the names of all the load balancers that are misconfigured.
+</Accordion>
+
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>